lib/core/proxy.c

```diff
@@ -225,6 +227,10 @@ static void build_request(h2o_req_t *req, h2o_iovec_t *method, h2o_url_t *url, h
                 } else if (token == H2O_TOKEN_EARLY_DATA) {
                     found_early_data = 1;
                     goto AddHeader;
+                } else if (token == H2O_TOKEN_EXPECT) {
+                    /* this never happens as protocol handlers should have consumed expect header */
+                    h2o_error_printf("[WARN] expect header unexpectedly found on http %x\n", req->version);
+                    continue;
```

Comment:
I'm not sure if we need to be worried specifically about the expect header getting through, even though we do not have these fail checks for other headers; e.g., Connection ,Upgrade, ...

Reply:
Addressed in https://github.com/h2o/h2o/pull/3316/commits/4f22cbc3c60c5a0363925db7026ab3ee574d693e